From 3b98f6c0f7193dcfaf52ccedd287220784ca90b2 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Fri, 10 Sep 2021 17:09:31 +0800
Subject: [PATCH 09/30] soc: rockchip: cpuinfo: Export rockchip_soc_id_init()
 function

Allow rockchip_soc_id_init() called before pure_initcall.

Change-Id: Ie0d3a18e96df02c2d6ab4aa3e17ea102685cd0c4
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
(cherry picked from commit f370f647892a4c1fd539c3b3eba5d53bede867be)
Reviewed-on: https://10.10.10.29/c/rk/kernel-stable/+/135013
Reviewed-by: Jianqun Xu <jay.xu@rock-chips.com>
Reviewed-by: Hans Yang <yhx@rock-chips.com>
Tested-by: Hans Yang <yhx@rock-chips.com>
---
 drivers/soc/rockchip/rockchip-cpuinfo.c | 5 ++++-
 include/linux/rockchip/cpu.h            | 7 +++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/rockchip/rockchip-cpuinfo.c b/drivers/soc/rockchip/rockchip-cpuinfo.c
index 2b55c8e..4199395 100644
--- a/drivers/soc/rockchip/rockchip-cpuinfo.c
+++ b/drivers/soc/rockchip/rockchip-cpuinfo.c
@@ -146,8 +146,11 @@ static void rk3308_init(void)
 	}
 }
 
-static int __init rockchip_soc_id_init(void)
+int __init rockchip_soc_id_init(void)
 {
+	if (rockchip_soc_id)
+		return 0;
+
 	if (cpu_is_rk3288()) {
 		rk3288_init();
 	} else if (cpu_is_rk312x()) {
diff --git a/include/linux/rockchip/cpu.h b/include/linux/rockchip/cpu.h
index 8db1293..debe1b8 100644
--- a/include/linux/rockchip/cpu.h
+++ b/include/linux/rockchip/cpu.h
@@ -36,6 +36,8 @@ static inline void rockchip_set_cpu_version(unsigned long ver)
 		(ver << ROCKCHIP_CPU_VERION_SHIFT) & ROCKCHIP_CPU_VERION_MASK;
 }
 
+int rockchip_soc_id_init(void);
+
 #else
 
 #define rockchip_soc_id 0
@@ -49,6 +51,11 @@ static inline void rockchip_set_cpu_version(unsigned long ver)
 {
 }
 
+static inline int rockchip_soc_id_init(void)
+{
+	return 0;
+}
+
 #endif
 
 #define ROCKCHIP_CPU_MASK       0xffff0000
-- 
2.7.4

